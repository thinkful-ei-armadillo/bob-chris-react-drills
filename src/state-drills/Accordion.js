import React, { Component } from 'react';


class Accordion extends Component{

    static defaultProps = {
        sections: []
    };

    state = { 
        buttonOpen: null
    }

    handleClicked = (e) => {
        this.setState({
            buttonOpen: e.currentTarget.getAttribute('id')
        })
        
    }

    render(){
        const cards = this.props.sections.map((section, index) =>
        {
            if (this.state.buttonOpen === index.toString()) {
                return (
                    <li key={index}><button id={index} onClick={this.handleClicked}>{section.title}</button>
                        <p>{section.content}</p>
                    </li>
                )
            }
            else {
                return <li key={index}><button id = {index} onClick = {this.handleClicked}>{section.title}</button></li>
            }
        })
        return(
            <ul>
                { cards }
            </ul>
        );
    }
    
}

export default Accordion