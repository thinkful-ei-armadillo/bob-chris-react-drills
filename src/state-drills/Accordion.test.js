import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import Accordion from './Accordion.js';
import toJson from 'enzyme-to-json'


describe('Testing Accordion app', () => {
  // smoke test

  const sections = [{ title: 'Section 1', content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.', },
  { title: 'Section 2', content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!', },
  { title: 'Section 3', content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?', },]

  it('renders empty given sections', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('The component renders no sections as active by default', () => {
    const wrapper = shallow(<Accordion sections = {sections}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('The component opens a clicked section', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('li').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
  it('only opens one section at a time', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('li').at(1).simulate('click')
    wrapper.find('li').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
});

/*
The component renders an empy li when the sections prop is not supplied
The component renders no sections as active by default
The component opens a clicked section
The component only opens the last section when multiple sections have been clicked.
*/