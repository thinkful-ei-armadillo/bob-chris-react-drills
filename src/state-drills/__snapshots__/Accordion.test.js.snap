// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Accordion app The component opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      id={0}
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      id={1}
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      id={2}
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Testing Accordion app The component renders no sections as active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      id={0}
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      id={1}
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      id={2}
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Testing Accordion app only opens one section at a time 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      id={0}
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      id={1}
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      id={2}
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Testing Accordion app renders empty given sections 1`] = `<ul />`;
